### PR TITLE
Stellar: _xdr_read_address now returns a string

### PR DESCRIPTION
**Pull request by zulucrypto**
Another minor change I found to make things compatible with the new message format that sends accounts as strings.

Also includes a bonus test for setting the inflation destination.

### CODE REVIEW

**Comment by zulucrypto** [comment]
@tsusanka - FYI I targeted this against your branch since it looked like it was still under development, let me know if it should be merged against a different one.

**Comment by tsusanka** [comment]
This is great! The commit https://github.com/trezor/python-trezor/commit/10bbb57c865ed514a01daa7b4e821207b62c7f89 got already merged to master though (from trezor:tsusanka/stellar-accounts-addresses), I think I just forgot to delete the branch, sorry about that.

Could you rebase on master and target against that?

**Comment by zulucrypto** [comment]
Rebased and target updated

**Comment by matejcik** [comment]
ah, so this is basically a bugfix for `trezorctl` -- otherwise parsing transactions would fail, right?

Would you mind adding a test for `parse_transaction_bytes`? Probably to `tests/unit_tests/stellar.py`, with a couple test vectors and assertions that the resulting protobuf objects are the right ones?

**Comment by zulucrypto** [comment]
@matejcik - Added unit tests for each operation (and caught a few more bugs in the process!)

File: trezorlib/stellar.py [hunk 1]
```diff
@@ -303,15 +303,15 @@ def _xdr_read_asset(unpacker):
...
+    """Reads a stellar address and returns the string representing the address
+    This method assumes the encoded address is a public address (starting with G)
     """
     # First 4 bytes are the address type
     address_type = unpacker.unpack_uint()
     if address_type != 0:
         raise ValueError("Unsupported address type")
 
-    return unpacker.unpack_fopaque(32)
+    return address_from_public_key(unpacker.unpack_fopaque(32))
```
Comment (matejcik) [inline trezorlib/stellar.py]: `address_from_public_key` still returns bytes though

File: trezorlib/tests/unit_tests/test_stellar.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import base64
+from trezorlib import stellar, messages as proto
```
Comment (matejcik) [inline trezorlib/tests/unit_tests/test_stellar.py]: please split into two lines:
```python
from trezorlib import messages
from trezorlib import stellar
```

also please change all usages of `proto` to `messages`. (ideally in stellar.py too)

File: trezorlib/tests/unit_tests/test_stellar.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    assert tx.sequence_number == 4294967296
+    assert tx.timebounds_start is None
+    assert tx.timebounds_end is None
+    assert tx.memo_type == stellar.MEMO_TYPE_NONE
+    assert tx.memo_text is None
+    assert tx.memo_id is None
+    assert tx.memo_hash is None
+    assert tx.num_operations == len(operations)
+
+def test_stellar_parse_transaction_bytes_memo_text():
```
Comment (matejcik) [inline trezorlib/tests/unit_tests/test_stellar.py]: please add an additional newline before every function, to appease `flake8`

File: trezorlib/tests/unit_tests/test_stellar.py [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+def test_stellar_parse_operation_bytes_bump_sequence_simple():
+    b64 = b'AAAAABXWSL/k028ZbPtXNf/YylTNS4Iz90PyJEnefPMBzbRpAAAAZAAAAAEAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAsAAAAASZYC0gAAAAA='
+
+    tx, operations = stellar.parse_transaction_bytes(base64.b64decode(b64))
+    op = operations[0]
+
+    assert isinstance(op, proto.StellarBumpSequenceOp)
+    assert op.source_account is None
+
+    assert op.bump_to == 1234567890
```
Comment (matejcik) [inline trezorlib/tests/unit_tests/test_stellar.py]: please add a newline at end of file

File: trezorlib/tests/unit_tests/test_stellar.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+    assert op.buying_asset.type == stellar.ASSET_TYPE_ALPHA4
+    # asset codes are either 4 or 12 characters, so this will be null-padded at the end
+    assert op.buying_asset.code == b'USD\x00'
+    assert op.buying_asset.issuer == b'GADEAA7R7K2HPZXIPQGUWCNLKQXUOQSMS6YZ6JQYO6DDLAQZZVIXG74A'
+
+    assert op.amount == 2000000000
+    assert op.price_n == 674614
+    assert op.price_d == 1000000
+    assert op.offer_id == 0 # indicates a new offer
```
Comment (matejcik) [inline trezorlib/tests/unit_tests/test_stellar.py]: another flake8 issue: there should be two spaces before the `#`

**Comment by matejcik** [comment]
@zulucrypto awesome, glad to see bugs being caught .)

there's one more bug that the test revealed to me: `address_from_public_key` returns bytes -- and you're testing for bytes in the tests. These should be strings, per PR title :)

this is going to be caught by type checking in the future, hopefully

**Comment by zulucrypto** [comment]
* Fixed flake8 errors
 * No longer aliasing `trezorlib.messages` to "proto"
 * Addresses are now utf-8 strings

**Comment by matejcik** [comment]
perfect, thank you.
merged via b2f35de8b8544c15c07503251e88b1eaa4f8703f

**Pull request closed without merging**